feat: add finders keepers algorithm to full stack (#61332)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lab-first-element-finder/a6e40f1041b06c996f7b2406.md

---
id: a6e40f1041b06c996f7b2406
title: Build a First Element Finder
challengeType: 26
dashedName: build-a-first-element-finder
---

> # --description--
> 
> In this lab, you will create a function that looks through an array and returns the first element in it that passes a 'truth test'.

Reviewer: ````suggestion
# --description--

In this lab, you will create a function that looks through an array and returns the first element that passes a test function (also known as a 'truth' test). 

The function whould iterate through the array and test each element using the provided test function. At the end, it would return the first element where the test function returns `true`.

Example:

```js
findElement([1, 3, 5, 8], num => num % 2 === 0) // returns 8
findElement([1, 3, 5], num => num % 2 === 0)    // returns undefined
```
````

---
id: a6e40f1041b06c996f7b2406
title: Build a First Element Finder
challengeType: 26
dashedName: build-a-first-element-finder
---

# --description--

> In this lab, you will create a function that looks through an array and returns the first element that passes a test function (also known as a 'truth' test). 

Reviewer 2: ```suggestion
In this lab, you will create a function that looks through an array and returns the first element that passes a test function (also known as a "truth test"). 
```

> The function whould iterate through the array and test each element using the provided test function. At the end, it would return the first element where the test function returns `true`.

Reviewer 2: ```suggestion
The function would iterate through the array and test each element using the provided test function. At the end, it would return the first element where the test function returns `true`.
```

Example:

```js
findElement([1, 3, 5, 8], num => num % 2 === 0) // returns 8
findElement([1, 3, 5], num => num % 2 === 0)    // returns undefined
```

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a function named `findElement` that accepts an array and a function as arguments.

> 2. The function should return the first item in the array that passes a 'truth test'. This means that, calling the passed in function `func`, given an element `x`, the 'truth test' is passed if `func(x)` is `true`.

Reviewer 2: I don't think quotes are necessary after the first time.
```suggestion
2. The function should return the first item in the array that passes a truth test. This means that, calling the passed in function `func`, given an element `x`, the truth test is passed if `func(x)` is `true`.
```

> 3. If no element passes the test, return `undefined`.

Reviewer 2: ```suggestion
3. If no element passes the test, the function should return `undefined`.
```

---
id: a6e40f1041b06c996f7b2406
title: Build a First Element Finder
challengeType: 26
dashedName: build-a-first-element-finder
---

# --description--

In this lab, you will create a function that looks through an array and returns the first element that passes a test function (also known as a "truth test"). 

The function would iterate through the array and test each element using the provided test function. At the end, it would return the first element where the test function returns `true`.

Example:

```js
findElement([1, 3, 5, 8], num => num % 2 === 0) // returns 8
findElement([1, 3, 5], num => num % 2 === 0)    // returns undefined
```

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a function named `findElement` that accepts an array and a function as arguments.
2. The function should return the first item in the array that passes a truth test. This means that, calling the passed in function `func`, given an element `x`, the truth test is passed if `func(x)` is `true`.
3. If no element passes the test, the function should return `undefined`.

> # --hints--

Reviewer 3: What about adding another tests that checks the number of parameters the function is defined with?

````md

`findElement` should take two parameters.

```js
assert.lengthOf(findElement, 2);
```

````

Reviewer: I am not sure it is necessary, the function can work without (`arguments` object, rest parameter...), and the other tests do not pass if the arguments are not used properly.

Here various situations that have the same result but one only of them would pass the test.

```js
function findElement(arr, func) {

// rest of the function

}
```

```js
function findElement() {
const arr = arguments[0];
const func = arguments[1];

// rest of the function

}
```

```js
function findElement(...args) {
const [arr, func] = args;
// rest of the function

}
```

challengeType: 26
dashedName: build-a-first-element-finder
---

# --description--

In this lab, you will create a function that looks through an array and returns the first element in it that passes a 'truth test'.

Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a function named `findElement` that accepts an array and a function as arguments.
2. The function should return the first item in the array that passes a 'truth test'. This means that, calling the passed in function `func`, given an element `x`, the 'truth test' is passed if `func(x)` is `true`.
3. If no element passes the test, return `undefined`.

# --hints--

`findElement([1, 3, 5, 8, 9, 10], function(num) { return num % 2 === 0; })` should return `8`.

```js
assert.strictEqual(
  findElement([1, 3, 5, 8, 9, 10], function (num) {
    return num % 2 === 0;
  }),
  8
);
```

`findElement([1, 3, 5, 9], function(num) { return num % 2 === 0; })` should return `undefined`.

```js
assert.strictEqual(
  findElement([1, 3, 5, 9], function (num) {
    return num % 2 === 0;
  }),
  undefined
);
```

# --seed--

## --seed-contents--

```js

> function findElement(arr, func) {

Reviewer 3: We are not usually providing code in the seed for labs.
